File: clientPython/zeroc-ice-3.7.2/js/test/typescript/Ice/servantLocator/TestI.ts
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import {Ice} from "ice";
import {Test} from "./generated";

export class TestI extends Test.TestIntf
{
    requestFailedException(current:Ice.Current):void
    {
    }

    unknownUserException(current:Ice.Current):void
    {
    }

    unknownLocalException(current:Ice.Current):void
    {
    }

    unknownException(current:Ice.Current):void
    {
    }

    localException(current:Ice.Current):void
    {
    }

    userException(current:Ice.Current):void
    {
    }

    jsException(current:Ice.Current):void
    {
        throw new Error();
    }

    javaException(current:Ice.Current):void
    {
    }

    unknownExceptionWithServantException(current:Ice.Current):void
    {
        throw new Ice.ObjectNotExistException();
    }

    impossibleException(throwEx:boolean, current:Ice.Current):string
    {
        if(throwEx)
        {
            throw new Test.TestImpossibleException();
        }

        //
        // Return a value so we can be sure that the stream position
        // is reset correctly if finished() throws.
        //
        return "Hello";
    }

    intfUserException(throwEx:boolean, current:Ice.Current):string
    {
        if(throwEx)
        {
            throw new Test.TestIntfUserException();
        }

        //
        // Return a value so we can be sure that the stream position
        // is reset correctly if finished() throws.
        //
        return "Hello";
    }

    asyncResponse(current:Ice.Current):void
    {
        throw new Ice.ObjectNotExistException();
    }

    asyncException(current:Ice.Current):void
    {
        throw new Ice.ObjectNotExistException();
    }

    shutdown(current:Ice.Current):void
    {
        current.adapter.deactivate();
    }
}

File: clientPython/zeroc-ice-3.7.2/java/test/src/main/java/test/Ice/interrupt/TestI.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package test.Ice.interrupt;

public class TestI implements test.Ice.interrupt.Test.TestIntf
{
    TestI(TestControllerI controller)
    {
        _controller = controller;
    }

    @Override
    public void op(com.zeroc.Ice.Current current)
    {
    }

    @Override
    public void opIdempotent(com.zeroc.Ice.Current current)
    {
        throw new com.zeroc.Ice.UnknownException();
    }

    @Override
    public void sleep(int to, com.zeroc.Ice.Current current)
        throws test.Ice.interrupt.Test.InterruptedException
    {
        _controller.addUpcallThread();
        try
        {
            Thread.sleep(to);
        }
        catch(InterruptedException ex)
        {
            throw new test.Ice.interrupt.Test.InterruptedException();
        }
        finally
        {
            _controller.removeUpcallThread();
        }
    }

    @Override
    public void opWithPayload(byte[] seq, com.zeroc.Ice.Current current)
    {
    }

    @Override
    public void shutdown(com.zeroc.Ice.Current current)
    {
        current.adapter.getCommunicator().shutdown();
    }

    private TestControllerI _controller;
}
